revisions.py

```diff
@@ -298,12 +306,36 @@ def rule_revision_batch(ctx, verbose):
             # Perform scheduled revision creation for one data object.
             data_id = row[0]
             path    = row[1] + "/" + row[2]
-            resc    = row[3]
+
+            test = int(hashlib.md5(path.encode('utf-8')).hexdigest(), 16) % 64 + 1
+            log.write(ctx, 'test hashlib: {}'.format(test))
+
+            # Metadata value contains resc and balace id for load balancing purposes.
+            info = row[3].split(',')
+            if len(info) == 2:
+                resc = info[0]
+                balance_id = int(info[1])
+            else:
+                # Backwards compatability with revision metadata created in v1.8 or earlier.
+                resc = row[3]
+                # Determine a balance_id for this dataobject based on its path.
+                # This will determine whether this dataobject will be taken into account in this job/range or another that is running parallel
+                balance_id = int(hashlib.md5(path.encode('utf-8')).hexdigest(), 16) % 64 + 1
+
+            # Check whether balance id is within the range for this job.
+            if balance_id < int(balance_id_min) or balance_id > int(balance_id_max):
+                # Skip this one and go to the next data object for revision creation.
+                if print_verbose:
+                    log.write(ctx, "Batch revision: skipping {} due to load balancing: range {} {}".format(path, balance_id_min, balance_id_max))
```

Comment:
I think this log message can be removed now that we have tested the load balancing.